Sources/TidepoolKit/TAPI.swift

```diff
@@ -728,6 +729,63 @@ public actor TAPI {
         let environment: String
     }
 
+    // MARK: - Device Logs
+
+    /// Uploads device logs
+    ///
+    /// - Parameters:
+    ///   - logs: entries to upload
+    ///   - start: the start date of the period that contains the log entries
+    ///   - end: the end date of the period that contains the log entries
+    public func uploadDeviceLogs(logs: [TDeviceLogEntry], start: Date, end: Date) async throws -> TDeviceLogsMetadata {
+        guard let session = session else {
+            throw TError.sessionMissing
+        }
+
+        var request = try createRequest(method: "POST", path: "v1/users/\(session.userId)/device_logs")
+        request.setValue("application/json; charset=utf-8", forHTTPHeaderField: HTTPHeaderField.contentType.rawValue)
+        let encoded = try JSONEncoder.tidepool.encode(logs)
+        request.httpBody = encoded
+
+        let digest = Insecure.MD5.hash(data: encoded)
+
+        let digestStr = "MD5=" + Data(digest).base64EncodedString()
+
+        print("Sending with start: \(start.timeString)")
+        request.setValue(start.timeString, forHTTPHeaderField: "X-Logs-Start-At-Time")
+        request.setValue(end.timeString, forHTTPHeaderField: "X-Logs-End-At-Time")
+
+        print("Sending with digest: \(digestStr)")
+        request.setValue(digestStr, forHTTPHeaderField: "Digest")
+
+        return try await performRequest(request)
+    }
+
+    /// List device logs metadata
+    ///
+    /// - Parameters:
+    ///   - logs: entries to upload
```

Comment:
`logs` comment should be removed.

Reply:
also return description should be added.

Reply:
Thanks!